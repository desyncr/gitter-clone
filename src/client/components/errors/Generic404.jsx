import React    from 'react';
import { Link } from 'react-router';
import               './errors.scss';

const Generic404 = () => (
  <div className="generic404">
    <h1>404</h1>
    <p>This is not the chat you are looking for,</p>
    <Link to="/explore">try exploring existing chats.</Link>
    <p>Or create a new chat</p>
  </div>
);

export default Generic404;
