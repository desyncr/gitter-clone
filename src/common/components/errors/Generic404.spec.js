import React from 'react';
import renderer from 'react-test-renderer';
import Generic404 from './Generic404';

const setup = () => {
  const component = <Generic404 />;

  return {
    component
  };
};

describe('<Generic404 />', () => {
  it('should render correctly', () => {
    const { component } = setup();
    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
