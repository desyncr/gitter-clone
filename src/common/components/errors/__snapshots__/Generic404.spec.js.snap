// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Generic404 /> should render correctly 1`] = `
<div
  className="generic404"
>
  <h1>
    404
  </h1>
  <p>
    This is not the chat you are looking for,
  </p>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    try exploring existing chats.
  </a>
  <p>
    Or create a new chat
  </p>
</div>
`;
